sympy/polys/matrices/sdm.py

```diff
@@ -970,3 +1012,34 @@ def sdm_particular_from_rref(A, ncols, pivots):
         if Ain is not None:
             P[j] = Ain / A[i][j]
     return P
+
+
+def sdm_applyfunc(M, shape, f, domain):
+    if domain.is_zero(f(domain.zero)):
+        return sdm_applyfunc_nonzero(M, f, domain)
+
+    nrows, ncols = shape
+    fzero = (f(domain.zero))
+    Mf = {i: {j: fzero for j in range(ncols)} for i in range(nrows)}
+
+    for i, Mi in M.items():
+        for j, Mij in Mi.items():
+                fMij = f(M[i][j])
+                if not domain.is_zero(fMij):
+                    Mf[i][j] = fMij
+                else:
+                    del Mf[i][j]
+    return Mf
+
+
+def sdm_applyfunc_nonzero(M, f, domain):
```

Comment:
There is already a function `unop_dict` that does this and can be used instead.

Reply:
Perhaps it could be renamed.

Reply:
Yeah maybe, I didn't know it was supposed to be doing the exact same thing as `sdm_applyfunc_nonzero`